cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java

```diff
@@ -148,6 +154,110 @@ void stop() throws SchedulerException {
   }
 
   @Override
+  public void addProgramSchedule(ProgramSchedule schedule) throws AlreadyExistsException, SchedulerException {
+    checkInitialized();
+    ProgramId program = schedule.getProgramId();
+    SchedulableProgramType programType = program.getType().getSchedulableType();
+
+    try {
+      checkScheduleExists(program, programType, schedule.getName());
+    } catch (org.quartz.SchedulerException e) {
+      throw new SchedulerException(e);
+    }
+
+    JobDetail job = addJob(program, programType);
+    scheduleJob(program, programType, schedule.getName(), ((TimeTrigger) schedule.getTrigger()).getCronExpression(),
+                job, schedule.getProperties());
+  }
+
+  @Override
```

Comment:
I feel like it may be simpler to keep the old `SchedulerService`, but we shouldn't call its methods for PartitionTrigger.